### PR TITLE
Allow running taker without maker online

### PR DESCRIPTION
**Pull request by klochowicz**
See commit messages for details.

### CODE REVIEW

**Comment by klochowicz** [comment]
image


(yes, I know it's not centered, I would like to try to bring the wallet balance back there - it looked nice with both inside, but I'd like ideally tackle the looks in a separate PR)

File: daemon/src/connection.rs [hunk 1]
```diff
@@ -172,3 +175,49 @@ impl Actor {
...
+pub async fn connect(
+    mut maker_online_status_feed_receiver: watch::Receiver<ConnectionStatus>,
+    connection_actor_addr: xtra::Address<Actor>,
+    maker_identity_pk: x25519_dalek::PublicKey,
+    maker_addresses: Vec<SocketAddr>,
+) {
+    loop {
+        if maker_online_status_feed_receiver.borrow().clone() == ConnectionStatus::Offline {
+            tracing::info!("No connection to the maker, attempting to connect:");
+            'reconnect: loop {
```
Comment (thomaseizinger) [inline daemon/src/connection.rs]: Is trying to connect to all resolved addresses really a reconnect?

File: daemon/src/connection.rs [hunk 2]
```diff
@@ -172,3 +175,49 @@ impl Actor {
...
 impl xtra::Actor for Actor {}
+
+pub async fn connect(
+    mut maker_online_status_feed_receiver: watch::Receiver<ConnectionStatus>,
+    connection_actor_addr: xtra::Address<Actor>,
+    maker_identity_pk: x25519_dalek::PublicKey,
+    maker_addresses: Vec<SocketAddr>,
+) {
+    loop {
+        if maker_online_status_feed_receiver.borrow().clone() == ConnectionStatus::Offline {
```
Comment (thomaseizinger) [inline daemon/src/connection.rs:188]: I would be nice if the reconnect logic _does not_ depend on this channel but is handled internally in the `connection::Actor`.

File: taker-frontend/src/components/NavBar.tsx [hunk 3]
```diff
@@ -20,9 +21,10 @@ import { WalletInfo } from "./Types";
 
 interface NavProps {
     walletInfo: WalletInfo | null;
+    makerStatus: boolean;
```
Comment (thomaseizinger) [inline taker-frontend/src/components/NavBar.tsx]: If this is a boolean, I would name it `connectedToMaker`.

**Review by thomaseizinger — APPROVED** [review]

File: daemon/src/connection.rs (hunk 2 above)
Comment (klochowicz) [reply daemon/src/connection.rs:188]: Indeed, that would be nicer - I'll aim to submit a  refactor PR after this one (to minimize the merge conflicts and allow to test the functionality in the meantime). Added a `TODO` for now.

File: daemon/src/connection.rs (hunk 1 above)
Comment (klochowicz) [reply daemon/src/connection.rs]: renamed to `connect`

File: taker-frontend/src/components/NavBar.tsx (hunk 3 above)
Comment (klochowicz) [reply taker-frontend/src/components/NavBar.tsx]: renamed it everywhere (most of the places had `isMakerOnline`, but I don't mind your version.

**Comment by klochowicz** [comment]
bors retry

**Pull request merged**